src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,115 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification.
+
+The typesystem is currently based on regular expressions. Each key is assigned with a regex
+that describes its contents. Links between keys can only be done if the regexes describing 
+the linked keys are compatible with each other.
+
+## Usage
+
+In order to use the type checker, mount a configuration specification along with this
+plugin. 
+
+`kdb mount <specification> spec/<path> <storage plugin to read the specification> typechecker`
+
+Currently there is no way to check the specification automatically upon mounting. Simply 
+retrieve a random key from the specification using `kdb get <path>` will cause
+the typechecking to happen, issuing a warning if it detects any problem with it.
+
+## Examples
+
+Create a sample configuration specification with three keys. As it is valid,
+there will be no error issued. Currently we include the type signatures of the
+various keywords manually in a specification. We use kdb shell to delay the 
+typechecking until we have finished writing the whole specification.
+
+```sh
+# Backup-and-Restore:spec/examples/simplespecification
+
+sudo kdb mount simplespecification.ini spec/examples/simplespecification ini typechecker
+
+echo 'kdbGet spec/examples/simplespecification \
+keySetName spec/examples/simplespecification/elektra/spec/fallback/# \
+keySetMeta elektra/spec/type RegexContains b a => Key b :: . -> Key a -> Key a \
```

Comment:
Is this something a user needs to write?

Reply:
right now he would have to or rather "copy the defaults", but i agree this is very bad and shouldn't be done by the user. i will think about a process to have a collection of type signatures for common operations like links or our checker plugins included with elektra so this would have only to be done when using custom unofficial plugins or similar.